docs/concepts/glossary.md

```diff
@@ -0,0 +1,140 @@
+The nodes are [Operators](./glossary#operator) and the edges indicate input/output-relationships or
+data streams or data sets.
+
+#### Managed State
+
+Managed State describes application state, which has been registered with the framework. For
+Managed State Apache Flink will take care about persistence and rescaling among other things.
+
+#### Operator
+
+Node of a [Logical Graph](./glossary#logical-graph). An Operator performs a certain operation,
+which is usually executed by a [Function](./glossary#function). Sources and Sinks are special
+Operators for data ingestion and data egress.
+
+#### Partition
+
+A partition is an independent subset of the overall data stream or data set. A data stream or
+data set is divided into partitions by assigning each [record](./glossary#Record) to one or more
+partitions. Partitions of data streams or data sets are consumed by [Tasks](./glossary#task) during
+runtime.
+
+#### Physical Graph
+
+A physical graph is the result of translating a [Logical Graph](./glossary#logical-graph) for
+execution in a distributed runtime. The nodes are [Tasks](./glossary#task) and the edges indicate
+input/output-relationships or partitions of data streams or data sets.
+
+#### Record
+
+Records are the constituent elements of a data set or data stream.
+[Operators](./glossary#operator) and [Functions](./glossary#Function) receive records as input
+and emit records as output.
+
+#### Flink Session Cluster
+
+A long-running [Flink Cluster](./glossary#flink-cluster), which accepts multiple
+[Flink Jobs](./glossary#flink-job) for execution. The lifetime of this Flink Cluster is not bound
+to the lifetime of any Flink Job. Formerly, a Flink Session Cluster was also known as a Flink Cluster in
+*session mode*.
+
+#### State Backend
+
+For stream processing programs the State Backend determines how state is stored on each Taskmanager
+(Java Heap of Taskmanager or RocksDB) as well as where it is written upon a checkpoint (Java Heap of Flink Master or
+Filesystem).
+
+#### Sub-Task
+
+A Sub-Task is a [Task](./glossary#task) responsible for processing a
+[partition](./glossary#partition) of the data stream. The term "Sub-Task" emphasizes that there are
+multiple parallel Tasks for the same [Operator](./glossary#operator).
+
+#### Task
+
+Node of a [Physical Graph](./glossary#physical-graph). A task is the basic unit of work, which is
+executed by Flink's runtime. Tasks usually encapsulate one or more [Operators](./glossary#operator).
+
+#### Transformation
+
+A transformation is applied on one or more data streams or data sets and results in one or more
+output new data streams or data sets. A transformation might change a data stream or data set on a
+per-record basis, but but might also only change its partitioning or perform an aggregation. In
+contrast to operations, transformations operate on streams or data sets, while operations operate
+on [records](./glossary#Record).
```

Comment:
I don't really get the last argument. Also, do you compare it with Operators, or Operations?

Reply:
For me, an operation would be the action that an operator performs. 

The idea is that transformations are applied on streams, while operations are applied on individual records.

Reply:
Can you give an example of a "transformation" and an "operation"? Maybe I'm mixing things up here...

Reply:
A transformation and and operation can do exactly the same thing. For example, a simple Map can be an operation if you talk about it on a per-record basis and a transformation if you talk about it on a stream.

For a repartitioning operations it makes a bit more sense, I think. A `keyBy` results is a transformation of the stream, but you can not repartition a single record. Of course, you could view it as a per-record operation, but it makes less sense, I think.

Reply:
That makes things clearer, but do we actually need the two different terms? I think, it may actually confuse people (like me) - unless more specifically explained as here.

Reply:
I changed this now. Since we don't mention the term "operation" anywhere, I removed this sentence and just noted that a transformation usually, but not always corresponds to an operator.